analyzers/src/SonarAnalyzer.Common/Extensions/AnalyzerOptionsExtensions.cs

```diff
@@ -33,6 +33,26 @@ public static class AnalyzerOptionsExtensions
     public static AdditionalText ProjectOutFolderPath(this AnalyzerOptions options) =>
         options.AdditionalFile("ProjectOutFolderPath.txt");
 
-    private static AdditionalText AdditionalFile(this AnalyzerOptions options, string fileName) =>
-        options.AdditionalFiles.FirstOrDefault(x => x.Path is not null && Path.GetFileName(x.Path).Equals(fileName, StringComparison.OrdinalIgnoreCase));
+    private static AdditionalText AdditionalFile(this AnalyzerOptions options, string fileName)
+    {
+        // HotPath: This code path needs to be allocation free. Don't use Linq.
+        foreach (var additionalText in options.AdditionalFiles) // Uses the struct enumerator of ImmutableArray
```

Comment:
💡 We would typically document this in Roslyn with the following attribute on the method:

```
[PerformanceSensitive(
    "https://github.com/SonarSource/sonar-dotnet/issues/7440",
    AllowCaptures = false,
    AllowGenericEnumeration = false)]
```

Reply:
It should be possible to use this attribute if you install the [Microsoft.CodeAnalysis.PerformanceSensitiveAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.PerformanceSensitiveAnalyzers) package

Reply:
Currently we don't have analyzer enforcement for the constraints of the attribute, but we hope to provide it in the future

Reply:
Thank you @sharwell for the hint.
I added #7690 to our backlog. We will discuss this internally.